### PR TITLE
`Development`: Add LTI profile 

### PR DESCRIPTION
**Pull request by krusche**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [X] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [X] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [X] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [X] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [X] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [X] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [X] I documented the TypeScript code using JSDoc style.

### Motivation and Context

We want to toggle the LTI feature through spring profiles.

### Description

We added the profile to all LTI related components and wrapped all LTI related functionality in checks for the profile or the corresponding components

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

File: src/main/webapp/app/course/manage/course-update.component.ts [hunk 1]
```diff
@@ -87,10 +86,12 @@ export class CourseUpdateComponent implements OnInit {
         private router: Router,
         private featureToggleService: FeatureToggleService,
         private accountService: AccountService,
-        private fb: FormBuilder,
     ) {}
 
     ngOnInit() {
+        this.profileService.getProfileInfo().subscribe((profileInfo) => {
+            this.ltiEnabled = profileInfo.activeProfiles.includes(PROFILE_LOCALVC);
+        });
```
Comment (julian-christl) [inline src/main/webapp/app/course/manage/course-update.component.ts:94]: Why don't you add this below? We already call the profile service in this onInit.

File: src/test/javascript/spec/component/course/course-update.component.spec.ts [hunk 2]
```diff
@@ -154,7 +154,7 @@ describe('Course Management Update Component', () => {
             expect(comp.courseOrganizations).toEqual([organization]);
             expect(getOrganizationsStub).toHaveBeenCalledOnce();
             expect(getOrganizationsStub).toHaveBeenCalledWith(course.id);
-            expect(getProfileStub).toHaveBeenCalledOnce();
+            expect(getProfileStub).toHaveBeenCalledTimes(2);
```
Comment (julian-christl) [inline src/test/javascript/spec/component/course/course-update.component.spec.ts:157]: Revert if you decide to remove the second call above.

File: src/main/java/de/tum/in/www1/artemis/web/rest/LegacyResource.java [hunk 3]
```diff
@@ -26,37 +19,15 @@
...
-     * POST lti/launch/:exerciseId : Launch the exercise app using request by an LTI consumer. Redirects the user to
-     * the exercise on success.
-     *
-     * @param launchRequest the LTI launch request (ExerciseLtiConfigurationDTO)
-     * @param exerciseId    the id of the exercise the user wants to open
-     * @param request       the request
-     * @param response      the response
-     * @deprecated use {@link PublicLtiResource#launch(LtiLaunchRequestDTO, Long, HttpServletRequest, HttpServletResponse)} instead
-     */
-    @PostMapping("lti/launch/{exerciseId}")
```
Comment (julian-christl) [inline src/main/java/de/tum/in/www1/artemis/web/rest/LegacyResource.java:51]: Are you aware that we mentioned in the release that this won't be removed before June 2024?

File: src/main/webapp/app/course/manage/course-update.component.ts (hunk 1 above)
Comment (krusche) [reply src/main/webapp/app/course/manage/course-update.component.ts:94]: just overlooked this, feel free to move it down

**Review by julian-christl — APPROVED** [review]

File: src/main/java/de/tum/in/www1/artemis/service/AssessmentService.java [hunk 4]
```diff
@@ -217,7 +217,9 @@ public Result submitManualAssessment(long resultId, Exercise exercise) {
         result.setCompletionDate(ZonedDateTime.now());
         result = resultRepository.submitResult(result, exercise, ExerciseDateService.getDueDate(result.getParticipation()));
         // Note: we always need to report the result (independent of the assessment due date) over LTI, otherwise it might never become visible in the external system
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/service/AssessmentService.java:219]: This Note seams to be outdated, since we don't *always* report it now

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

**Review by JohannesStoehr — APPROVED** [review]
Approve code

**Review by Strohgelaender — APPROVED** [review]
code looks good

**Pull request merged**